hw/ip/prim/doc/prim_flash.md

```diff
@@ -1,157 +1,159 @@
+  {name: 'prog_last_i', wave: '0.......1....0...'},
+  {name: 'ack_o',       wave: '010..10.....10...'},
+  {name: 'done_o',      wave: '0..............10'},
+]}
+{{< /wavejson >}}
+
+### Erase
+{{< wavejson >}}
+{signal: [
+  {name: 'clk_i',     wave: 'p................'},
+  {name: '*_erase_i', wave: '01.0.........1.0.'},
+  {name: 'ack_o',     wave: '0.10..........10.'},
+  {name: 'done_o',    wave: '0.....10.........'},
+]}
+{{< /wavejson >}}
+
+## Initlialization
+
+The flash wrapper may undergo technology specific intializations when it is first powered up.
+During this state, it asserts the `init_busy` to inform the outside world that it is not ready for transactions.
+During this time, if a transaction is issued towards the flash wrapper, the transaction is not acknowledged until the initialization is complete.
+
+## Program Beats
+
+Since flash programs can take a significant amount of time, certain flash wrappers employ methods to optimize the program operation.
+This optimization may place an upper limit on how many flash words can be handled at a time.
+The purpose of the `prog_last` is thus to indicate when a program burst has completed.
+
+Assume the flash wrapper can handle 16 words per program operation.
+Assume a program burst has only 15 words to program and thus will not fill up the full program resolution.
+On the 15th word, the `prog_last` signal asserts and informs the flash wrapper that it should not expect a 16th word and should proceed to complete the program operation.
+
+## Program Type
+The `prog_type` input informs the flash wrapper what type of program operation it should perform.
+A program type not supported by the wrapper, indicated through `prog_type_avail` shall never be issued to the flash wrapper.
+
+## Erase Suspend
+Since erase operations can take a significant amount of time, sometimes it is necessary for software or other components to suspend the operation.
+The suspend operation input request starts with `erase_suspend_req` assertion. Flash wrapper circuit acks when wrapper starts suspend. 
+When the erase suspend completes, the flash wrapper circuitry also asserts `done` for the ongoing erase transaction to ensure all hardware gracefully completes.
+
+The following is an example diagram
+{{< wavejson >}}
+{signal: [
+  {name: 'clk_i',                wave: 'p................'},
+  {name: 'pg_erase_i',           wave: '01.0..............'},
+  {name: 'ack_o',                wave: '0.10...10........'},
+  {name: 'erase_suspend_i',      wave: '0.....1.0........'},
+  {name: 'done_o',               wave: '0............10..'},
+ ]
+  }
+{{< /wavejson >}}
+
+## Error Interrupt
+The `flash_err_o` is a level interrupt indication, that is asserted whenever an error event occurs in one of the Flash banks. An Error status register is used to hold the error source of both banks, and it is cleared on read. Clearing the status register trigs deassertion of the interrupt.
```

Comment:
would you mind clarifying if the `flash_err` signal is pulse or level? This is mostly just for integration purposes.

Reply:
You mean signal name should be modified? I will change it to become flash_err_lv_o, ok?